### PR TITLE
Create game night

### LINKED ISSUE #2: Add GameNight End-Point
- [x] Check GameNight db, model, and tests

Add attributes:
- [x] GN name
- [x] Date
- [x] Number of games
- [x] List of users to add to invitations table

From dupe ticket:
- [x] Add relationship between game night and user/invitees
- [x] Add relationship between games of users attached to game night. (this might not be a strict relationship) 

Right now the GameNight knows it's host, but it also needs to have a relationship with invitees (has many invitations) and games (little more difficult as you have to get it through the user/host and the invitation users).

### LINKED ISSUE #5: Gane Night Show Page Endpoint
- [ ]  return invitees, games owned by all invitees that haven't declined, and data about game night. 
- [ ] Collate the games from the invitees and de-dupe

### PR DESCRIPTION
**Pull request by katemorris**
### Description

This PR created two endpoints, create a game night and access a single game night. 

### Closes issue(s)

#2 #5 

### Testing
Needs some sad path soon but I am worn out right now. We can add those after talking about how to send back errors. 

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [X] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [X] I ran rubocop and fixed issues
- [X] I ran full test suite - all tests passing 

### Other comments

The attendees are sent back with all data, that needs to be restricted to name and email but the FriendsSerializer wasn't working inside of the GameNightSerializer. Alas.

### CODE REVIEW

File: app/serializers/game_night_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  attribute :games do |game_night|
+    game_night.games_to_play.each do |game|
+      game
+    end
+  end
+
+  # Need to figure out how to just send name and email.
+  attribute :attendees do |game_night|
+    game_night.attendees.each do |attendee|
+      attendee
```
Comment (PhilipDeFraties) [inline app/serializers/game_night_serializer.rb:14]: I think on line 14 you could just call attendee.name, attendee.email

**Review by PhilipDeFraties — APPROVED** [review]

File: config/routes.rb [hunk 2]
```diff
@@ -16,7 +16,12 @@
         get :friends, to: 'users/friends#index'
       end
 
-      resources :games, only: [:create]
+      # Games crud
+      resources :games do
+        post :games, to: 'games/games#create'
+      end
```
Comment (taphill) [inline config/routes.rb]: I think we want to keep the `resources :games, only: [:create]` route here and get rid of

```
resources :games do
  post :games, to: 'games/games#create'
end
```

That seemed to be creating a lot of routes that we didn't need unless I'm missing something?

**Review by taphill — CHANGES_REQUESTED** [review]
Left some feedback regarding the routes...let me know what you think!

File: config/routes.rb (hunk 2 above)
Comment (katemorris) [reply config/routes.rb]: I wasn't sure which one was best! Sorry about that! I'll edit and get that pushed back up in just a bit.

Comment (katemorris) [reply config/routes.rb]: Updated!

**Review by taphill — APPROVED** [review]

**Pull request merged**